src/wp-admin/includes/file.php

```diff
@@ -2557,3 +2639,65 @@ function wp_opcache_invalidate( $filepath, $force = false ) {
 
 	return false;
 }
+
+/**
+ * Attempts to clear the opcode cache for a directory of files.
+ *
+ * @since 6.2.0
+ *
+ * @see wp_opcache_invalidate()
+ * @link https://www.php.net/manual/en/function.opcache-invalidate.php
+ * @global WP_Filesystem_Base $wp_filesystem WordPress filesystem subclass.
+ *
+ * @param string $dir The path to the directory for which the opcode cache is to be cleared.
+ */
+function wp_opcache_invalidate_directory( $dir ) {
+	global $wp_filesystem;
+
+	if ( ! is_string( $dir ) || '' === trim( $dir ) ) {
+		$error_message = sprintf(
+			/* translators: %s: The '$dir' argument. */
+			__( 'The %s argument must be a non-empty string.' ),
+			'<code>$dir</code>'
+		);
+		// phpcs:ignore WordPress.PHP.DevelopmentFunctions.error_log_trigger_error
+		trigger_error( wp_kses_post( $error_message ) );
```

Comment:
Not sure `wp_kses_post()` is needed here? This is a hard-coded string/message, what would KSES do here?

Reply:
The variable might need to be escaped as it would come from the user. Or am I incorrect and no escaping is needed.

Reply:
> it would come from the user

How? It can only come from core or plugins/themes. If some plugin decides to accept user input, it should verify/filter it before passing to this (or any other) function.

The second reason is that this code runs when `$dir` is not a string. Instead of `wp_kses_post()` it should probably be `strval()` or `"$dir"` of it may throw errors (for example what if it is array?). 

And the third reason is that KSES is fairly slow and intended only for user input before it is saved to the DB. It's no good to use it on display, ever, even for short strings.

Reply:
I was thinking if someone might call that function directly and potentially be malicious. If you disagree I'm happy to remove it or change it.

Perhaps the simpler path is to not return `$dir` with the error message?

Reply:
or perhaps

```php
		$error_message = sprintf(
			/* translators: %s: The function. */
			__( 'The argument for %s must be a non-empty string.' ),
			__FUNCTION__
		);
		// phpcs:ignore WordPress.PHP.DevelopmentFunctions.error_log_trigger_error
		trigger_error( $error_message );
```

Reply:
Actually, if `$dir` is not a string don't think it is a good idea to include it in the PHP notice. 

Frankly starting to wonder if always throwing a PHP error would really be helpful here... Would probably be but only in "development mode". Probably best to do only when `if ( WP_DEBUG )`?

Reply:
So something like this?

```php
	if ( ! is_string( $dir ) || '' === trim( $dir ) ) {
		$error_message = sprintf(
			/* translators: %s: The function. */
			__( 'The argument for %s must be a non-empty string.' ),
			__FUNCTION__
		);
		if ( defined( 'WP_DEBUG' ) && WP_DEBUG ) {
			// phpcs:ignore WordPress.PHP.DevelopmentFunctions.error_log_trigger_error
			trigger_error( $error_message );
		}
		return;
	}
```

Reply:
Uh, sorry, that's weird. Didn't see your latest comments while I was posting...

Reply:
> something like this?

Generally there's no need to do `defined( 'WP_DEBUG' )`. It's always defined in https://github.com/WordPress/wordpress-develop/blob/trunk/src/wp-includes/default-constants.php#L81.

Reply:
PR updated

Reply:
I was incorrect. We don't need any sort of escaping as`$dir` is between single quotes and not double quotes so cannot be interpreted.